pkg/controller/deployment/deployment_controller.go

```diff
@@ -1294,3 +1413,27 @@ func (dc *DeploymentController) rollbackToTemplate(deployment *extensions.Deploy
 	d, err = dc.updateDeploymentAndClearRollbackTo(deployment)
 	return
 }
+
+// isScalingEvent checks whether the provided deployment has been updated with a scaling event
+// by looking at the desired-replicas annotation in the active replica sets of the deployment.
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
+	if err != nil {
+		return false
+	}
+	allRSs := append(oldRSs, newRS)
+	for _, rs := range controller.FilterActiveReplicaSets(allRSs) {
+		if desiredString, ok := rs.Annotations[deploymentutil.DesiredReplicasAnnotation]; ok {
```

Comment:
Hm, it's impossible to add custom methods on replica sets. Potentailly, I could have a helper to do this.

> What does it mean when this is not 'ok'? It looks like the code interprets this as 'the desired matches the deployment'

It means that either the replica set is adopted and the deployment controller hasn't observed it yet or someone is messing with those annotations which is not good (for them). In any case just the size of a replica set that lacks this annotation cannot help us determine if a scaling event just happened.